### ISSUE TITLE
Xpetra: Turn off warnings for deprecated code

### ISSUE THREAD

**Issue by maxfirmbach** [trilinos/Trilinos#13250]
## Question
@trilinos/xpetra @mayrmt 

Currently one can set the following flag `Trilinos_SHOW_DEPRECATED_WARNINGS=OFF` to avoid all the warnings related to the deprecation of `Epetra`. In addition one can set the flag `Xpetra_ENABLE_DEPRECATED_CODE=ON` to enable the `Epetra` backend of `Xpetra`, which again results in deprecation warnings, which are not caught by `Trilinos_SHOW_DEPRECATED_WARNINGS`.

Would it be possible to also supress these warnings (Epetra related in Xpetra) with `Trilinos_SHOW_DEPRECATED_WARNINGS`?

**Comment by mayrmt** [comment]
@trilinos/framework

**Comment by mayrmt** [comment]
The warnings have been introduced by @sebrowne via PR #12828.

**Comment by sebrowne** [comment]
I could have sworn I tested that case and it did disable the errors.  I will see if I can figure out what is going on today.  Those warnings ARE intended to respect the `Trilinos_SHOW_DEPRECATED_WARNINGS` value.

**Comment by sebrowne** [comment]
The warnings that we added are not displayed (correctly).  I do see some additional ones that were displayed that look like this:
```
In file included from Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_CrsMatrixFactory.hpp:59,
                 from Trilinos/packages/xpetra/sup/Matrix/Xpetra_CrsMatrixWrap_def.hpp:63,
                 from Trilinos/build/packages/xpetra/sup/Utils/ExplicitInstantiation/Xpetra_CrsMatrixWrap.cpp:55:
Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp:74:2: warning: #warning is a GCC extension
 #warning "The header file Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp is deprecated."
  ^~~~~~~
Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp:74:2: warning: #warning "The header file Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp is deprecated." [-Wcpp]
```

@maxfirmbach were these the warnings you wanted hidden behind the `${packageName}_SHOW_DEPRECATED_WARNINGS` option?  I can add that conditional if you'd like.

**Comment by maxfirmbach** [comment]
@sebrowne Yes, exactly that are the ones :sweat_smile:

**Comment by maxfirmbach** [comment]
> The warnings that we added are not displayed (correctly). I do see some additional ones that were displayed that look like this:
> 
> ```
> In file included from Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_CrsMatrixFactory.hpp:59,
>                  from Trilinos/packages/xpetra/sup/Matrix/Xpetra_CrsMatrixWrap_def.hpp:63,
>                  from Trilinos/build/packages/xpetra/sup/Utils/ExplicitInstantiation/Xpetra_CrsMatrixWrap.cpp:55:
> Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp:74:2: warning: #warning is a GCC extension
>  #warning "The header file Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp is deprecated."
>   ^~~~~~~
> Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp:74:2: warning: #warning "The header file Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp is deprecated." [-Wcpp]
> ```
> 
> @maxfirmbach were these the warnings you wanted hidden behind the `${packageName}_SHOW_DEPRECATED_WARNINGS` option? I can add that conditional if you'd like.

@sebrowne It would be nice if they could be turned off in some way, even an aditional `Xpetra_SHOW_DEPRECATED_WARNINGS` would be fine :+1:

**Comment by sebrowne** [comment]
> @sebrowne Will the fix be part of release 16.0?

No, unless it is highly important I do not want to hold the 16.0.0 release any longer.  We can do a 16.1.0 if it's high priority to get that fix included.

@maxfirmbach that's exactly how it will work, but that variable propagates from `Trilinos_SHOW_DEPRECATED_WARNINGS`, so it should work either way.  I'll do my best to get it in this evening.

**Comment by mayrmt** [comment]
@sebrowne I totally understand.

Well, we'd appreciate a timely 16.1.0 release, so that we can update to an official release version soon.

**Comment by sebrowne** [comment]
PR merged and labeled as being added to the 16.1 release notes
